src/pages/RoomInvitePage.tsx

```diff
@@ -121,57 +105,55 @@ function RoomInvitePage(props) {
-                const accountID = lodashGet(option, 'accountID', null);
-                const isOptionInPersonalDetails = _.some(personalDetails, (personalDetail) => personalDetail.accountID === accountID);
+            filterSelectedOptions = selectedOptions.filter((option) => {
+                const accountID = option?.accountID;
+                const isOptionInPersonalDetails = invitePersonalDetails.some((personalDetail) => accountID && personalDetail.accountID && personalDetail.accountID === accountID);
                 const parsedPhoneNumber = parsePhoneNumber(LoginUtils.appendCountryCode(Str.removeSMSDomain(searchTerm)));
-                const searchValue = parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : searchTerm.toLowerCase();
-                const isPartOfSearchTerm = option.text.toLowerCase().includes(searchValue) || option.login.toLowerCase().includes(searchValue);
-                return isPartOfSearchTerm || isOptionInPersonalDetails;
+                const searchValue = parsedPhoneNumber.possible && parsedPhoneNumber.number ? parsedPhoneNumber.number.e164 : searchTerm.toLowerCase();
+                const isPartOfSearchTerm = option.text?.toLowerCase().includes(searchValue) ?? option.login?.toLowerCase().includes(searchValue);
+                return isPartOfSearchTerm ?? isOptionInPersonalDetails;
             });
         }
+        const filterSelectedOptionsFormatted = filterSelectedOptions.map((selectedOption) => OptionsListUtils.formatMemberForList(selectedOption));
 
         sectionsArr.push({
             title: undefined,
-            data: filterSelectedOptions,
-            shouldShow: true,
+            data: filterSelectedOptionsFormatted,
             indexOffset,
         });
         indexOffset += filterSelectedOptions.length;
 
         // Filtering out selected users from the search results
-        const selectedLogins = _.map(selectedOptions, ({login}) => login);
-        const personalDetailsWithoutSelected = _.filter(personalDetails, ({login}) => !_.contains(selectedLogins, login));
-        const personalDetailsFormatted = _.map(personalDetailsWithoutSelected, (personalDetail) => OptionsListUtils.formatMemberForList(personalDetail, false));
-        const hasUnselectedUserToInvite = userToInvite && !_.contains(selectedLogins, userToInvite.login);
+        const selectedLogins = selectedOptions.map(({login}) => login);
+        const personalDetailsWithoutSelected = invitePersonalDetails.filter(({login}) => !selectedLogins.includes(login));
+        const personalDetailsFormatted = personalDetailsWithoutSelected.map((personalDetail) => OptionsListUtils.formatMemberForList(personalDetail));
+        const hasUnselectedUserToInvite = userToInvite && !selectedLogins.includes(userToInvite.login);
 
         sectionsArr.push({
             title: translate('common.contacts'),
             data: personalDetailsFormatted,
-            shouldShow: !_.isEmpty(personalDetailsFormatted),
```

Comment:
why this was removed?

Reply:
shouldShow is not used anywhere in SelectionList(sectionArr is data for SectionList) so I've removed it.
Double checked if everything works.